core/src/test/java/org/elasticsearch/search/aggregations/bucket/adjacency/InternalAdjacencyMatrixTests.java

```diff
@@ -37,22 +39,24 @@
     @Override
     public void setUp() throws Exception {
         super.setUp();
-        keys = new ArrayList<>();
+        List<String> listOfKeys = new ArrayList<>();
         int numFilters = randomIntBetween(2, 4);
```

Comment:
The number of buckets should be limited to 10 (4x4 matrix, but we only represent the upper triangular matrix). I would like it better if the maxNumberOfBuckets() was somehow tied to the number of filters we generate in this particular test. There are only three possibilities: 3 (for a 2x2 matrix), 6 (3x3 matric) and 10. Is allowing for max of 10 here possible (the AggregationsTests setup overwrite is okay I think)